code/game/turfs/open/lava.dm

```diff
@@ -345,6 +345,7 @@
 	if(!plasma_parts.len && !robo_parts.len) //a person with no potential organic limbs left AND no robotic limbs, time to turn them into a plasmaman
 		burn_human.ignite_mob()
 		burn_human.set_species(/datum/species/plasmaman)
+		burn_human.adjustToxLoss(-burn_human.getToxLoss()) // clear the toxin damage after turning into a plasmaman
```

Comment:
wouldn't the better fix here be making gaining the trait that makes you immune to tox damage clear your toxloss (and/or not block tox healing)

Reply:
Hmm. Yeah, I could do that.